src/components/Sidebar.js

```diff
@@ -1,13 +1,72 @@
-import { Link } from 'react-router-dom';
+import { useState } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import { Icon } from '@iconify/react';
 
 export default function Sidebar() {
+  const [open, setOpen] = useState(true);
+
   return (
-    <div className="w-60 p-4 flex-shrink-0 shadow-xl">
-      <h1 className="font-semibold text-2xl mb-8">Coverage Checker</h1>
-      <nav className="flex flex-col gap-2">
-        <Link to="/file-upload">Upload Files</Link>
-        <Link to="/">mCODE Coverage</Link>
-      </nav>
+    <div className={`flex flex-col ${open ? 'w-60' : 'w-20'} p-4 h-screen sticky top-0 flex-shrink-0 shadow-xl`}>
+      {/* sidebar header */}
+      <div className="flex flex-row">
+        <h1 className={`${open ? '' : 'hidden'} font-semibold text-xl mb-8`}>Coverage Checker</h1>
+        <Icon
+          icon="akar-icons:chevron-down"
+          className="ml-2"
+          height="24"
+          rotate="90deg"
+          vFlip={!open}
+          onClick={() => setOpen(!open)}
+        />
+      </div>
+      {/* sidebar body */}
+      <div className="flex-1">
+        {/* upper section */}
+        <div className="h-3/5 flex flex-col gap-5">
+          <h2 className="font-semibold text-base mb-2">{open ? 'Navigation' : 'Nav'}</h2>
```

Comment:
Let's add a new color to our tailwind config, `#767676`, and use that color here. We can call it `heading-grey`.